deluge 2.0.3 and libtorrentRasterbar 1.2.5

petabyteboy opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Alternative to https://github.com/NixOS/nixpkgs/pull/64496 and https://github.com/NixOS/nixpkgs/pull/64542

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

dtzWill committed:
deluge: 1.3.15 -> 2.0.3

* let's try 2.0 version now, no time better than the present! Maybe!
* bz2 -> xz
* maybe python3
* disable pyGtkGlade for deps, maybe not needed?
* fix gtk/etc deps, deluge-gtk works! \o/
* restore installation of images and such

The old version is kept available as some torrent trackers have not
updated their whitelists yet.

dtzWill committed:
libtorrent-rasterbar: 1.1.11 -> 1.2.5

https://raw.githubusercontent.com/arvidn/libtorrent/libtorrent-1_2_5/ChangeLog

The old release is kept available as libtorrentRasterbar-1_1_x for deluge 1.x.

dtzWill committed:
deluge: remove use of deprecated alias

dtzWill commented:
Thank you! Sorry for not following-up on my PR :cat2: .

On that thread there was discussion from some folks experiencing missing icons (quite possibly fixed with a rebase or changes you've made), so hopefully that's not a problem still.... but the other detail was ensuring the NixOS service module is fixed/updated accordingly.  I don't know anything about that.

As a start, let's see what running the test does:

@GrahamcOfBorg test deluge

petabyteboy commented:
Tests are not working currently.
Icon issue is still there when deluge is run from nix-shell, I will have a look.
I hope we can make it so that users can just decide what deluge version to use by setting a package option. Not sure if it is necessary to run them in parallel.

petabyteboy commented:
I think the best option for the service might be to use the stateVersion to determine the default package. This is because upgrading changes the application state irreversably, so we want users who are currently using 1.x to make a conscious choice on that. 
On the other hand we want new users to get the latest version by default.

peterhoeg commented:
The nixos module requires some serious work for deluge 2.

I have a WIP branch here https://github.com/peterhoeg/nixpkgs/commits/u/deluge that might help you.

bricewge commented on pkgs/development/libraries/libtorrent-rasterbar/1.2/default.nix:
```diff
@@ -0,0 +1,52 @@
+
+  enableParallelBuilding = true;
+  nativeBuildInputs = [ automake autoconf libtool pkgconfig ];
+  buildInputs = [ boostPython openssl zlib python libiconv geoip ncurses ];
```
`geoip` has been removed from libtorrent-rasterbar since 1.1: https://github.com/arvidn/libtorrent/blob/55460a60bc722f27d1300bddd9743d4d59492da7/include/libtorrent/session_handle.hpp#L550

Why the uses of `ncurses`? I can't find any reference to it in their repo?

petabyteboy replied:
ncurses is required for the configure checks to succeed when building with python3. I have removed geoip from both versions and confirmed in the output binaries that nothing changed.

bricewge commented on pkgs/development/libraries/libtorrent-rasterbar/1.2/default.nix:
```diff
@@ -0,0 +1,52 @@
+    "--with-libiconv=yes"
+    "--with-boost=${boostPython.dev}"
+    "--with-boost-libdir=${boostPython.out}/lib"
+    "--with-libiconv=yes"
```
`"--with-libiconv=yes"` duplicated, libtorrent-rasterbar-1.1 has the same typo.

petabyteboy replied:
Removed, thanks

svalaskevicius commented:
Do the toolbar icons appear in this pr? I had to add adwaita icon theme for them in my pr (I didn't see this one before)

svalaskevicius commented:
Oh and hicolor icon theme as well for the logs to stop spamming about missing icons

petabyteboy commented:
Does anyone still need to run deluge 1.x? I just rechecked all my trackers and it seems like they all support deluge 2.x now, so I have no need for running the old version.

peterhoeg commented:
The nixos module also needs to support this before we can merge it.

petabyteboy commented:
That's why I'm asking, there is still significant work to be done to support both versions in parallel, which would be unnecessary if noone is using it.
But yes, the declarative configuration for deluge 2.0 something I have not taken a closer look yet, but I will do.

peterhoeg commented:
> That's why I'm asking, there is still significant work to be done to support both versions in parallel, which would be unnecessary if noone is using it.

If I were you, I definitely wouldn't try supporting both in tandem considering how much has changed - it would be a ton of work for very little gain.

mweinelt commented:
Icons are working as soon as I add `gnome3.adwaita-icon-theme` to my environment. That fixes icons in a lot of other apps and according to https://nixos.wiki/wiki/GNOME we are on our own to fix them if we're not actually using GNOME.

Commit:
nixos/deluge: support 2.x

petabyteboy commented:
Fixed module and test for deluge 2.x while preserving support for deluge 1.x

petabyteboy commented:
Do we want to fix the icon thing? I think I saw a comment somewhere that users are expected to have one icon theme in their environment according to the manual, so this would not strictly be a bug.

mweinelt commented:
I don't think so, according to the GNOME wikipage this is something
everyone as to decide for themselves. I guess if we wanted that changed
we'd need to open up an issue on the matter.

On 3/19/20 4:18 AM, Milan wrote:
> Do we want to fix the icon thing? I think I saw a comment somewhere that
> users are expected to have one icon theme in their environment according
> to the manual, so this would not strictly be a bug.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/77762#issuecomment-600966243>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABAEDYBZRAB6TCPDMGEIMDRIGFJLANCNFSM4KHDNJCQ>.
>

petabyteboy commented:
I moved this to my overlay and am using it on all my machines. From my side this is ready to be merged.

peterhoeg commented:
@petabyteboy, do you mind fixing up the merge conflicts so we can get this merged?

petabyteboy commented:
Feel free to do so. I'm running this since two weeks. Other than concerns about data migration it should be good to go. The topic of data migration is solved sufficiently for me, but others might have different standards.

peterhoeg commented:
@bricewge raised some issues though.

Would also be great to hear from maintainers @domenkozar @ebzzry

petabyteboy commented:
Oh, I will address those comments now.

petabyteboy commented:
So while working on the comments I also rebased it to master and noticed that something was not quite right. It turns libtorrentRasterbar was already updated in #83414, and deluge 1.3.15 continues to build fine with the new version. This is a surprise to me, but well, let's leave it that way then.

Commit:
Partially revert "libtorrent-raster: 1.1.11 -> 1.2.5; qbittorrent: 4.2.1 -> 4.2.2"

This partially reverts commit cc03fb4210c472e7ed172079af0a4016228bbee1.
The libtorrentRasterbar update broke deluge 1.x, the hash was not
updated and obsolete dependencies and flags were not removed.

petabyteboy commented:
Rebased, squashed together the libtorrentRasterbar updates and updated it to 1.2.5.

peterhoeg commented:
With the state check I think it's safe to merge despite the user having to migrate from 1.x to 2.x, *but* we definitely need a few lines in the release notes.

petabyteboy commented:
I added a note to the release notes, I hope I found the correct section (it is called backwards incompatible but most changes described there are more like forward incompatible).

petabyteboy commented:
@worldofpeace can you tell me if the release note is ok like this? I would like to merge this.

worldofpeace commented:
@petabyteboy Note looks good and is in the correct section. Seems fine to me, let's go :+1:

petabyteboy commented:
Thanks everyone :sparkles:
